### DISCUSSION TITLE
/Storage rights are set RO, why?

### DISCUSSION THREAD

**Discussion by RHKCommander959** [jlesage/docker-handbrake#295] [General]
So at work we are using Unraid with docker and this fork of handbrake. On an older version I could use the GUI just fine, but newer versions changed the /storage permissions, so I can read file contents but not process/save anything. Manually launching the image with the permissions changed to RW allow it to work just fine though. We also use the watch and output functions just fine to automate converting videos, but the GUI is nice for one-offs.

So what gives, why is /storage set for RO now? Is this a security issue, or a typo, or what? Unfortunately with Unraid if you launch customized images you can't directly edit them like you can if you use the default settings, nor get updates.

**Comment by jlesage** [comment]
The unRAID template always had `/storage` RO by default.  But this is something that can be changed afterward (once the container is installed).

The reasoning behind that is that `/storage` is where your data reside, it is your "input" or "source" folder.  Usually people use `/output` to store the converted videos.  You don't want to provide R/W permissions if they are not required.  This prevents any possibility for the container to mess with your data.

**Comment by RHKCommander959** [comment]
Thanks for the information and fast response!

So taking a look at it again today, handbrake is trying to use /storage by
default for GUI manual output, not /output. The default destination is
/storage, and trying to change it on the GUI at the bottom right still
references /storage instead of /output.

I'm certainly not an unraid pro, but I'm not seeing anywhere to change
permissions, nor changing /output to be the default versus /storage.

I've tried chmod as well  no luck. So far the only route to get it to work
for me is setting RW for /storage on a custom container.

**Comment by jlesage** [comment]
> and trying to change it on the GUI at the bottom right still
references /storage instead of /output.

Not sure what you mean.  What happen when you select `/output` ?

**Reply by RHKCommander959** [reply]
There is no option to select /output, only /storage. I can try to get some screenshots if it helps?

**Reply by jlesage** [reply]
Yes, that would help to make sure we are on the same page.
